### PR TITLE
[GHA] Switched python nightly tests to self-hosted runners back

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Fixed crashing issue (error 255) and  switched python tightly tests to self-hosted runners back 

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by rkazants — CHANGES_REQUESTED** [review]
Please check that nightly run is not affected due to low power machines used for this.
Is timeout acceptable for nightly run?

File: tests/constraints.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ fastjsonschema~=2.17.1
 tensorflow>=2.5,<2.15.0
 test-generator==0.1.2
 requests>=2.25.1
-opencv-python>=4.5
+opencv-python-headless>=4.5
```
Comment (rkazants) [inline tests/constraints.txt:17]: why do we need to use headless version?

Comment (mryzhov) [reply tests/constraints.txt]: it was a test only) installation of opencv-python-headless should take less time. But I saw errors, reverted

**Comment by mryzhov** [comment]
> Please check that nightly run is not affected due to low power machines used for this. Is timeout acceptable for nightly run?
Tested here: 
https://github.com/openvinotoolkit/openvino/pull/20992

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -1270,19 +1270,18 @@ jobs:
...
-    # container:
-    #   image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04
-    #   volumes:
-    #     - /mount/caches:/mount/caches
+    timeout-minutes: ${{ github.event_name == 'schedule' && 400 || 10 }}
+    runs-on: ${{ github.event_name == 'schedule' && 'aks-linux-8-cores-32gb' || 'aks-linux-8-cores-16gb'}}
+    container:
+      image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04
+      volumes:
+        - /mount/caches:/mount/caches
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:1278]: let's do the same for conformance

**Review by rkazants — CHANGES_REQUESTED** [review]
I see no complete results for nightly. Please check https://github.com/openvinotoolkit/openvino/actions/runs/6813088336?pr=20992

**Pull request closed without merging**